File: galaxy-qa-backoffice/src/main/java/fr/lovotech/galaxy/qa/backoffice/domain/DataTestCaseVersion.java
package fr.lovotech.galaxy.qa.backoffice.domain;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

import java.io.Serializable;
import java.util.Date;

@AllArgsConstructor
@NoArgsConstructor
@Data
@Document(collection = "dataTestCaseVersion")
@Builder(toBuilder = true)
public class DataTestCaseVersion extends AbstractEntity implements Serializable {
    private static final long serialVersionUID = 1L;

    private ReferenceUID externalUid;
    private Date creationDate;
    @DBRef
    private DataTestCase dataTestCase;
    @DBRef
    private ApplicationVersion applicationVersion;
    private FileInfo fileInfo;
    private Date endDate;
}

File: galaxy-test-fo/src/app/shared/models/dataTestCaseVersion.ts
import { ApplicationVersion } from "./application-version";
import { DataTestCase } from "./dataTestCase";
import { FileInfo } from "./FileInfo";
import { Test } from "./test";

export interface DataTestCaseVersion{
 applicationVersion?:ApplicationVersion;
 dataTestCase?:DataTestCase;
 fileInfo?:FileInfo;
}
